src/login/pam_systemd.c

```diff
@@ -251,7 +253,13 @@ static int get_seat_from_display(const char *display, const char **seat, uint32_
         if (r < 0)
                 return r;
 
-        r = get_ctty(ucred.pid, NULL, &tty);
+        r = get_ctty_devnr(ucred.pid, &display_ctty);
+        if (r < 0)
+                return r;
+
+        if (asprintf(&sys_path, "/sys/dev/char/%d:%d", major(display_ctty), minor(display_ctty)) < 0)
+                return -ENOMEM;
+        r = readlink_value(sys_path, &tty);
         if (r < 0)
```

Comment:
hmm, is this is sufficient this way? I mean `tty` is now going to be something like `../../devices/virtual/tty/tty4/`, but vtnr_from_tty() cannot parse this, can it? what am I missing?

Reply:
Doesn't `readlink_value()` return only the basename (`tty4`), which `vtnr_from_tty()` should accept? But I'll rebuild and retest in case I used old version in testing.

Reply:
Also rebuilt version works just fine.
```
lightdm[1195]: pam_systemd(lightdm:session): pam-systemd initializing
lightdm[1195]: pam_systemd(lightdm:session): Asking logind to create session: uid=1000 pid=1195 service=lightdm type=x11 class=user desktop=plasma seat=seat0 vtnr=1 tty= display=:0 remote=no remote_user= remote_host=

$ ps -up 1145 1195
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
xorg        1145  4.3  0.6 231388 103520 tty1    Ssl+ 16:48   0:12 /usr/lib/xorg/Xorg :0 vt1 -br -nolisten tcp -logverbose -100000 -verbose 1 -background none -noreset -seat seat0 -logfile /dev/null -keeptty -once
root        1195  0.0  0.0 152004  8424 ?        Sl   16:48   0:00 lightdm --session-child 15 20
```
In my setup, lightdm does not start Xorg, instead it is started as standalone systemd service. This lets me harden Xorg sandboxing separately from lightdm / user session.